python/pyspark/sql/conf.py

```diff
@@ -35,12 +35,14 @@ def __init__(self, jconf: JavaObject) -> None:
         self._jconf = jconf
 
     @since(2.0)
-    def set(self, key: str, value: str) -> None:
+    def set(self, key: str, value: Union[str, int, bool]) -> None:
         """Sets the given Spark runtime configuration property."""
```

Comment:
if we reuse existing doc in sql, we normally add a comment like this:

```
    .. versionchanged:: 3.4.0
        Support Spark Connect.
```

or maybe
```
    .. versionchanged:: 4.0.0 (or 3.5.0?)
        Support Spark Connect.
```

Reply:
If this gets merged soon, I think we can just mark it as 3.4.0.